File: frontend/src/lib/types/pac.ts
export interface PACMes {
  mes: number;
  valor_programado: number;
}

export interface PACResumenRubro {
  codigo: string;
  cuenta: string;
  apropiacion_definitiva: number;
  total_pac: number;
  pac_configurado: boolean;
  pac: PACMes[];
}

export interface PACUpdate {
  valores_mensuales: number[];
}

export interface PACDisponible {
  codigo_rubro: string;
  mes: number;
  valor_programado: number;
  pagos_mes: number;
  disponible: number;
}

File: app/schemas/pac.py
from pydantic import BaseModel


class PACMesResponse(BaseModel):
    mes: int
    valor_programado: float

    model_config = {"from_attributes": True}


class PACUpdate(BaseModel):
    valores_mensuales: list[float]  # 12 values, one per month


class PACResumenRubroResponse(BaseModel):
    codigo: str
    cuenta: str
    apropiacion_definitiva: float
    total_pac: float
    pac_configurado: bool
    pac: list[PACMesResponse]

    model_config = {"from_attributes": True}


class PACDisponibleResponse(BaseModel):
    codigo_rubro: str
    mes: int
    valor_programado: float
    pagos_mes: float
    disponible: float
